src/inmanta/logging.py

```diff
@@ -608,7 +631,36 @@ def apply_options(self, options: Options, component: str | None = None, context:
                 self.force_cli(convert_inmanta_log_level(str(options.verbose), cli=True))
         else:
             self._apply_logging_config_from_options(options, component, context)
-        self._options_applied = True
+
+        self._options_applied = options
+        self._component = component
+        self._context = context
+
+    def extend_config(self, extenders: "list[LoggingConfigBuilderExtension]" = []) -> FullLoggingConfig:
```

Comment:
Mutable defaults are bad practice.
```suggestion
    def extend_config(self, extenders: "list[LoggingConfigBuilderExtension]" = None) -> FullLoggingConfig:
```